### PR TITLE
feat(sdk): add support for OKX wallet

### PR DESCRIPTION
**Pull request by danadi7**
#### What this PR does / why we need it:
- Adds support for OKX wallet

#### Which issue(s) does this PR fixes?:

Fixes #ORD-87

### CODE REVIEW

File: packages/sdk/src/browser-wallets/okx/index.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  const provider = await getOKXWalletProvider(network);
+
+  const { address, publicKey } = await provider.connect();
```
Comment (Nanosync) [inline packages/sdk/src/browser-wallets/okx/index.ts:48]: I think we will need to add try-catch on this line to ensure that we throw request cancelled by user error

File: packages/sdk/src/browser-wallets/okx/index.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import { OKXSignPSBTOptions } from "./types";
+
+function isInstalled(): boolean {
+  if (typeof window === "undefined") {
+    throw new OrditSDKError("Cannot call this function outside a browser");
+  }
+  return typeof window.okxwallet !== "undefined";
+}
+
+async function getOKXWalletProvider(
```
Comment (Nanosync) [inline packages/sdk/src/browser-wallets/okx/index.ts:20]: Async can be removed for this function as there are no promises used

File: packages/sdk/src/browser-wallets/okx/index.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  const provider = await getOKXWalletProvider(network);
```
Comment (Nanosync) [inline packages/sdk/src/browser-wallets/okx/index.ts:46]: We can remove await here

Comment (danadi7) [reply packages/sdk/src/browser-wallets/okx/index.ts:46]: Removed!

File: packages/sdk/src/browser-wallets/okx/index.ts (hunk 2 above)
Comment (danadi7) [reply packages/sdk/src/browser-wallets/okx/index.ts:20]: Oh yes. Changed!

File: packages/sdk/src/browser-wallets/okx/index.ts (hunk 1 above)
Comment (danadi7) [reply packages/sdk/src/browser-wallets/okx/index.ts:48]: Added error handling for all methods

**Review by Nanosync — APPROVED** [review]

File: packages/sdk/src/browser-wallets/okx/index.ts [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    signedPsbtHex = await provider.signPsbt(psbtHex, {
+      autoFinalized: finalize,
+      toSignInputs,
+    });
+  } catch (err) {
+    const okxError = err as OKXError;
+    if (okxError.code === 4001) {
+      throw new BrowserWalletRequestCancelledByUserError();
+    }
+    throw new OrditSDKError(okxError.message);
```
Comment (Nanosync) [inline packages/sdk/src/browser-wallets/okx/index.ts:128]: Should we propagate the okx error? If it's empty, should we let it fall through to line 131?

File: packages/sdk/src/browser-wallets/okx/index.ts [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+  let signature: string = "";
+
+  try {
+    signature = await provider.signMessage(message, type);
+  } catch (err) {
+    const okxError = err as OKXError;
+    if (okxError.code === 4001) {
+      throw new BrowserWalletRequestCancelledByUserError();
+    }
+    throw new OrditSDKError(okxError.message);
```
Comment (Nanosync) [inline packages/sdk/src/browser-wallets/okx/index.ts:182]: same question here, https://github.com/ordzaar/ordit-sdk/pull/151/files#r1520834083

**Pull request merged**